<a href="https://colab.research.google.com/github/tigzx9/emerging/blob/main/Class02.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Simple Python


In the class A, students' heights are:

161.5
165.9
168.7
163.1
170.8
167.2

In the class B, students' heights are:

172.5
162.9
161.7
160.1
161.8
169.2

The class A and the class B are joined together to form the class AB. Write a code to calculate the mean height by using Python 'list' and 'for' loop

In [1]:
# Define the heights of students in class A and class B
class_a_heights = [161.5, 165.9, 168.7, 163.1, 170.8, 167.2]
class_b_heights = [172.5, 162.9, 161.7, 160.1, 161.8, 169.2]

# Combine the heights of the two classes
class_ab_heights = class_a_heights + class_b_heights

# Calculate the mean height using a for loop
total_height = 0
for height in class_ab_heights:
    total_height += height
    print(height)
mean_height = total_height / len(class_ab_heights)

# Print the mean height
print("The mean height of class AB is:", mean_height)

161.5
165.9
168.7
163.1
170.8
167.2
172.5
162.9
161.7
160.1
161.8
169.2
The mean height of class AB is: 165.45000000000002


Write a python code to sort the heights of the class AB.

 The class C is formed with students with the height less than 165. Make a list 'C' containing the heights of the class C.

The class D is formed with students with the height larger than 160 and less than 170. Make a list containing the heights of the class D.

Use a lambda function and map() to answer the following questions.
1). Add 1 to each element of the list 'a' and create another list.

a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

In the following list of names, extract only those stargin with 'J'. Show the result as a list.

a = ['Sara', 'Jack', 'Rebecca', 'Jennifer', 'Paul', 'Andrew', 'John']

Code the class named 'Calculator' that outputs the result as shown.

calc1 = Calculator(5,3)


print(calc1.sum())

print(calc1.subtract())

print(calc1.product())

print(calc1.divide())


8

2

15

1.6666666666666667

Write function named 'palindrome' to test whether the entered word or sentence is a palindrome.
Ignore capitalization, spaces, punctuation marks, symbols, etc. This code should be written with simple python without importing any libraries

Palindrome examples:

mom

wow

noon

level

radar

kayak

racecar

God's dog

Madam, I'm Adam

#Example from pandas, numpy, and matplotlib libraries

Write a python code that generates 1000 random number between -5 and 5 with normal distribution and then plot them:

Read in the data:

!wget --no-clobber https://raw.githubusercontent.com/tn-220/EmergingClass/main/data_studentlist.csv

df = pd.read_csv('data_studentlist.csv', header='infer')

df.head(5)

**Answer the following questions using Pandas library.**

1) Average height of the male students:

In [27]:
import pandas as pd

!wget --no-clobber https://raw.githubusercontent.com/tn-220/EmergingClass/main/data_studentlist.csv
# Read the dataset
df = pd.read_csv('data_studentlist.csv', header='infer')

# Filter out the male students
male_df = df[df['gender'] == 'M']

# Compute the mean height of the male students
mean_height_male = male_df['height'].mean()

print("Average height of male students: {:.2f} cm".format(mean_height_male))

File ‘data_studentlist.csv’ already there; not retrieving.

Average height of male students: 172.41 cm


2) Average height of the female students:

In [28]:
import pandas as pd

!wget --no-clobber https://raw.githubusercontent.com/tn-220/EmergingClass/main/data_studentlist.csv
# Read the dataset
df = pd.read_csv('data_studentlist.csv', header='infer')

# Filter out the male students
male_df = df[df['gender'] == 'F']

# Compute the mean height of the male students
mean_height_male = male_df['height'].mean()

print("Average height of female students: {:.2f} cm".format(mean_height_male))

File ‘data_studentlist.csv’ already there; not retrieving.

Average height of male students: 166.64 cm


3) Average weight of the male students:

In [29]:
import pandas as pd

!wget --no-clobber https://raw.githubusercontent.com/tn-220/EmergingClass/main/data_studentlist.csv
# Read the dataset
df = pd.read_csv('data_studentlist.csv', header='infer')

# Filter out the male students
male_df = df[df['gender'] == 'M']

# Compute the mean height of the male students
mean_height_male = male_df['weight'].mean()

print("Average weight of male students: {:.2f} kg".format(mean_height_male))

File ‘data_studentlist.csv’ already there; not retrieving.

Average weight of male students: 68.50 kg


4) The tallest height among the male students:

In [30]:
import pandas as pd

!wget --no-clobber https://raw.githubusercontent.com/tn-220/EmergingClass/main/data_studentlist.csv
# Read the dataset
df = pd.read_csv('data_studentlist.csv', header='infer')

# Filter out the male students
male_df = df[df['gender'] == 'M']

# Compute the mean height of the male students
mean_height_male = male_df['height'].max()

print("Max height of male students: {:.2f} cm".format(mean_height_male))

File ‘data_studentlist.csv’ already there; not retrieving.

Max height of male students: 182.10 cm


5) The average grade of students with absence ('Y'):


In [31]:
import pandas as pd

!wget --no-clobber https://raw.githubusercontent.com/tn-220/EmergingClass/main/data_studentlist.csv
# Read the dataset
df = pd.read_csv('data_studentlist.csv', header='infer')

# Filter out the male students
male_df = df[df['absence'] == 'Y']

# Compute the mean height of the male students
mean_height_male = male_df['grade'].mean()

print("Average height of male students: {:.2f} cm".format(mean_height_male))

File ‘data_studentlist.csv’ already there; not retrieving.

Average height of male students: 3.00 cm


6) Average age of the students with absence ('Y') whose grade is equal or larger than 3:

In [36]:
import pandas as pd

!wget --no-clobber https://raw.githubusercontent.com/tn-220/EmergingClass/main/data_studentlist.csv
# Read the dataset
df = pd.read_csv('data_studentlist.csv', header='infer')

# Filter out the male students
male_df = df[df['absence'] == 'Y']
print(male_df)
age_df = male_df[male_df['grade'] >= 3]
print(age_df)
# Compute the mean height of the male students
mean_height_male = male_df['age'].mean()

print("Average height of male students: {:.2f} cm".format(mean_height_male))

File ‘data_studentlist.csv’ already there; not retrieving.

               name gender  age  grade absence bloodtype  height  weight
0     Jared Diamond      M   23      3       Y         O   165.3    68.2
4   Clara Rodriquez      F   20      1       Y         A   168.0    49.5
10    Joshua Connor      M   24      4       Y         O   167.1    62.0
15   Richard Swayze      M   24      4       Y         B   168.6    70.2
              name gender  age  grade absence bloodtype  height  weight
0    Jared Diamond      M   23      3       Y         O   165.3    68.2
10   Joshua Connor      M   24      4       Y         O   167.1    62.0
15  Richard Swayze      M   24      4       Y         B   168.6    70.2
Average height of male students: 22.75 cm


Use pandas and numpy for following:

Read in the data.

!wget --no-clobber https://raw.githubusercontent.com/tn-220/EmergingClass/main/data_census.csv

df = pd.read_csv('data_census.csv', header='infer')

df.shape

df.head(10)

**Answer the following questions using Pandas library.**

1) Total population of ProvinceCode equal to 115 and 116:

2) Average population of the cities where there are more males (GenderRatio >1:


3) Places with more males (GenderRatio >1) and less than 2 persons per household:

 


4) Sort the DataFrame in ascending order of the 'Households'. Show the top 10.

###Import pandas, numpy , and mathplotlib

Read in the data.
!wget --no-clobber https://raw.githubusercontent.com/tn-220/EmergingClass/main/data_coffeeshop.csv

**Answer the following questions.**

1) Make a frequency table of 'yearOfStart' and visualize by year.


* Sort by the year.

* Draw a line plot from 1997 to 2014. <= Hint: plt.xlim()

2) Now, split the data by the current state of business ('In' or 'Out' of business). Then, visualize the yearly trend of the 'yearOfStart' frequencies.

* Sort by the year.
* Draw two overlapping line plots from 1997 to 2014.
* Use the 'figure' object.

3) From the results of 1) and 2), how can you describe the trend in general?

* Around the year 2008, began a steep increase in the number of new coffee shops.
* As of the year 2014, more coffee shops opened in the past 5 years are still in business rather than going out of business.